lisa/analysis/tasks.py

```diff
@@ -423,6 +423,36 @@ def df_task_total_residency(self, task):
 
         return residency_df
 
+    @df_task_states.used_events
+    def df_tasks_total_residency(self, tasks=None, ascending=False):
+        """
+        DataFrame of tasks execution time on each CPU
+
+        :param tasks: List of tasks to report, all trace tasks by default
+        :type tasks: list(int) or list(str) or a mix of the two
+
+        :param ascending: Set True to plot before short running tasks,
+                          False by deafult
+        :type ascending: bool
+        """
+        if tasks is None:
+            tasks = list(self.trace.get_tasks().keys())
+        res_df = pd.DataFrame()
+        for task in tasks:
+            if isinstance(task, str):
+                pid = self.trace.get_task_by_name(task)
+            else:
+                pid = int(task)
+                task = self.trace.get_task_by_pid(pid)
+                mapping = {'runtime': '{}:[{}]'.format(pid, task)}
+                _df = self.trace.analysis.tasks.df_task_total_residency(pid).T.rename(index=mapping)
+                res_df = res_df.append(_df)
```

Comment:
So the above 3 lines will only execute if the task is an int. If you pass a string this will not be executed. It seems to me there's an indentation error here? Otherwise you're processing things in the else block only.

Reply:
Right, that has broken in one of the last push :/